tests/api/retrieval.test.js

```diff
@@ -0,0 +1,262 @@
+      method: "GET",
+      headers: {
+        "Content-Type": "application/json",
+        Origin: "http://localhost:3000/api/retrieval?maxRecords=",
+        authorization: `Bearer ${token}`,
+      },
+      query: {
+        maxRecords: "",
+      },
+    });
+
+    executeQuery.mockImplementation((client, sql) => {
+      if (sql.includes("SELECT bearer_token as bearerToken FROM templates")) {
+        return {
+          rows: [{ bearerToken: token }],
+          rowCount: 1,
+        };
+      }
+    });
+    await retrieval(req, res);
+    expect(res.statusCode).toBe(400);
+    expect(JSON.parse(res._getData())).toEqual(
+      expect.objectContaining({ error: "Invalid payload value found maxRecords" })
+    );
+  });
+
+  it("Should return 400 status code with an invalid formID found ", async () => {
+    const mockSession = {
+      expires: "1",
+      user: { email: "forms@cds.ca", name: "forms" },
+    };
+    const token = jwt.sign(
+      {
+        formID: undefined,
+      },
+      process.env.TOKEN_SECRET,
+      {
+        expiresIn: "1y",
+      }
+    );
+
+    client.getSession.mockReturnValue(mockSession);
+
+    const { req, res } = createMocks({
+      method: "GET",
+      headers: {
+        "Content-Type": "application/json",
+        Origin: "http://localhost:3000/api/retrieval?maxRecords=1",
+        authorization: `Bearer ${token}`,
+      },
+      query: {
+        maxRecords: "1",
+      },
+    });
+
+    executeQuery.mockImplementation((client, sql) => {
+      if (sql.includes("SELECT bearer_token as bearerToken FROM templates")) {
+        return {
+          rows: [{ bearerToken: token }],
+          rowCount: 1,
+        };
+      }
+    });
+
+    await retrieval(req, res);
+    expect(res.statusCode).toBe(404);
+    expect(JSON.parse(res._getData())).toEqual(expect.objectContaining({ error: "Bad Request" }));
+  });
+
+  it("Should return a list of form responses with 200 status code", async () => {
+    const mockSession = {
+      expires: "1",
+      user: { email: "forms@cds.ca", name: "forms" },
+    };
+    const token = jwt.sign(
+      {
+        formID: 1,
+      },
+      process.env.TOKEN_SECRET,
+      {
+        expiresIn: "1y",
+      }
+    );
+    client.getSession.mockReturnValue(mockSession);
+    const { req, res } = createMocks({
+      method: "GET",
+      headers: {
+        "Content-Type": "application/json",
+        Origin: "http://localhost:3000/api/retrieval?maxRecords=2",
+        authorization: `Bearer ${token}`,
+      },
+      query: {
+        maxRecords: "2",
+      },
+    });
+
+    executeQuery.mockImplementation((client, sql) => {
+      if (sql.includes("SELECT bearer_token as bearerToken FROM templates")) {
+        return {
+          rows: [{ bearerToken: token }],
+          rowCount: 1,
+        };
+      }
+    });
+    const dynamodbExpectedReponses = {
+      Items: [
+        { FormID: "1", SubmissionID: "12", FormSubmission: true },
```

Comment:
Is that how Dynamo returns responses or is there an extra layer of "S" and "N" to denote types. The unit test should probably reflect this so its as close to real life as possible